c_api/tests/c_api_behavior_test.cpp

```diff
@@ -113,3 +113,453 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
 
   inner(TI_ARCH_VULKAN);
 }
+
+TEST_F(CapiTest, TestBehaviorAllocateMemory) {
+  TiError error = TI_ERROR_SUCCESS;
+  TiMemoryAllocateInfo allocate_info;
+  allocate_info.size = 1024;
+  auto inner = [&](TiArch arch) {
+    if (ti::is_arch_available(arch)) {
+      // Attempt to allocate memory with size of 1024
+      TiRuntime runtime = ti_create_runtime(arch);
+      for (int i = 0; i < 4; ++i) {
+        allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT << i;
```

Comment:
Check if it sets a `TI_ERROR_ARGUMENT_OUT_OF_RANGE` if usage is `TI_MEMORY_USAGE_MAX_ENUM`.

Reply:
In this case you only need `TI_MEMORY_USAGE_STORAGE_BIT` checked. That's the only flag required by Taichi Runtime. The others are for the underlying RHI. See the documentation for detail: https://docs.taichi-lang.org/docs/master/taichi_core#bitfield-timemoryusageflags.

For texture flags, check `TI_IMAGE_USAGE_STORAGE_BIT` and `TI_IMAGE_USAGE_SAMPLED_BIT`.